linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java

```diff
@@ -0,0 +1,143 @@
+  @ApiImplicitParams({
+    @ApiImplicitParam(name = "emInstance", required = true, dataType = "String"),
+    @ApiImplicitParam(name = "instance", required = true, dataType = "String"),
+    @ApiImplicitParam(name = "logDirSuffix", required = true, dataType = "String"),
+    @ApiImplicitParam(name = "logType", required = true, dataType = "String")
+  })
+  @ApiOperationSupport(ignoreParameters = {"json"})
+  @RequestMapping(path = "/downloadEngineLog", method = RequestMethod.GET)
+  public void downloadEngineLog(
+      HttpServletRequest req,
+      HttpServletResponse response,
+      @RequestParam(value = "emInstance") String emInstance,
+      @RequestParam(value = "instance") String instance,
+      @RequestParam(value = "logDirSuffix") String logDirSuffix,
+      @RequestParam(value = "logType") String logType)
+      throws IOException, ECMErrorException {
+    ModuleUserUtils.getOperationUser(req, "downloadEngineLog");
+    if (StringUtils.isBlank(instance)) {
+      throw new ECMErrorException(
+          PARAMETER_NOT_NULL.getErrorCode(),
+          MessageFormat.format(PARAMETER_NOT_NULL.getErrorDesc(), "instance"));
+    }
+    if (StringUtils.isBlank(logDirSuffix)) {
+      throw new ECMErrorException(
+          PARAMETER_NOT_NULL.getErrorCode(),
+          MessageFormat.format(PARAMETER_NOT_NULL.getErrorDesc(), "logDirSuffix"));
+    }
+    if (StringUtils.isBlank(logType)) {
+      throw new ECMErrorException(
+          PARAMETER_NOT_NULL.getErrorCode(),
+          MessageFormat.format(PARAMETER_NOT_NULL.getErrorDesc(), "logType"));
+    }
+    File inputFile = new File(logDirSuffix, logType);
+    if (!inputFile.exists()) {
+      throw new ECMErrorException(
+          LOG_IS_NOT_EXISTS.getErrorCode(),
+          MessageFormat.format(LOG_IS_NOT_EXISTS.getErrorDesc(), logDirSuffix));
+    } else {
+      long fileSizeInBytes = inputFile.length();
+      long fileSizeInMegabytes = fileSizeInBytes / (1024 * 1024);
+      if (fileSizeInMegabytes > 100) {
+        throw new ECMErrorException(
+            FILE_IS_OVERSIZE.getErrorCode(),
+            MessageFormat.format(FILE_IS_OVERSIZE.getErrorDesc(), logDirSuffix));
+      }
+      ServletOutputStream outputStream = null;
+      FileInputStream inputStream = null;
+      BufferedInputStream fis = null;
+      PrintWriter writer = null;
+      try {
+        inputStream = new FileInputStream(inputFile);
+        fis = new BufferedInputStream(inputStream);
+        byte[] buffer = new byte[1024];
+        int bytesRead = 0;
+        response.setCharacterEncoding(Consts.UTF_8.toString());
+        java.nio.file.Path source = Paths.get(inputFile.getPath());
+        response.addHeader("Content-Type", Files.probeContentType(source));
+        response.addHeader(
+            "Content-Disposition",
+            "attachment;filename=" + instance.replace(":", "_") + "_" + logType + ".txt");
+        response.addHeader("Content-Length", fileSizeInBytes + "");
```

Comment:
add annotation for filename example
// filename eg:xxxxx
val filename= instance.replace(":", "_") + "_" + logType + ".txt"
.....

Reply:
done